test/basic.test.js

```diff
@@ -1,14 +1,29 @@
 'use strict'
 const os = require('os')
 const { join } = require('path')
-const { readFileSync } = require('fs')
+const { readFileSync, existsSync } = require('fs')
 const { test } = require('tap')
 const { sink, check, once } = require('./helper')
 const pino = require('../')
 const { version } = require('../package.json')
 const { pid } = process
 const hostname = os.hostname()
-const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms))
+const watchFileCreated = (filename, timeout) => new Promise((resolve, reject) => {
+  const interval = setInterval(() => {
+    if (existsSync(filename)) {
+      clearTimeout(timer)
+      clearInterval(interval)
+      resolve()
+    }
+  }, 100)
+  const timer = setTimeout(() => {
+    clearTimeout(timer)
+    clearInterval(interval)
+    reject(new Error(`${filename} was not created with a Timeout:${timeout}`))
+  },
+  timeout
+  )
```

Comment:
Would you mind removing the timer and just add a counter in the interval?
we can also move the timeout to a constant inside this function.

Reply:
I've pushed a change. I thinks it fit your review, or may i've misunderstood 🙃